packages/headless/src/controllers/insight/sort/headless-insight-sort.ts

```diff
@@ -0,0 +1,39 @@
+import {executeSearch} from '../../../features/insight-search/insight-search-actions';
+import {SortCriterion} from '../../../features/sort-criteria/criteria';
+import {logResultsSort} from '../../../features/sort-criteria/sort-criteria-insight-analytics-actions';
+import {InsightEngine} from '../../../app/insight-engine/insight-engine';
+import {
+  buildCoreSort,
+  Sort,
+  SortProps,
+  SortState,
+  SortInitialState,
+} from '../../core/sort/headless-core-sort';
+
+export type {Sort, SortProps, SortState, SortInitialState};
+
+/**
+ * Creates an `InsightSort` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `InsightSort` controller properties.
+ * @returns A `InsightSort` controller instance.
+ */
```

Comment:
I'd suggest to mention the `Sort` interface in the comments to match the function signature.